core/src/main/java/org/apache/iceberg/actions/BaseRewriteDataFilesAction.java

```diff
@@ -228,9 +228,18 @@ public RewriteDataFilesActionResult execute() {
         .flatMap(Streams::stream)
         .collect(Collectors.toList());
 
+    // add a filter  to the CombinedScanTask list to avoid repeated rewrite datafile
+    List<CombinedScanTask> fileterCombinedScanTasks =
+        combinedScanTasks.stream().filter(task -> task.files().size() > 1).collect(Collectors.toList());
```

Comment:
Maybe more clear as !(task.files.nonempty)

Reply:
Could also push this into the build of Combined scan tasks on line 228, rather than doing a new variable

Reply:
> Maybe more clear as !(task.files.nonempty)

we need to remove the combinedScanTasks which size is 1 ，not `nonempty`

Reply:
> Could also push this into the build of Combined scan tasks on line 228, rather than doing a new variable

I refer to the variable filteredGroupedTasks above，maybe it is more clear

Reply:
I agree with @RussellSpitzer 's comment about "push this into the build of Combined scan tasks on line 228", because in that way we don't have to construct the `List<CombinedScanTask>` twice,  that is unnecessary. 

BTW,  would you pls add few unit tests to address this case ?  I think it's great to improve this but better to fix it with a UT.

Thanks for the work.

Reply:
I update the pr，add a test  case